src/main/java/seedu/address/model/person/ArgumentPredicate.java

```diff
@@ -64,35 +52,37 @@ private void addIfTagPresent(Set<Tag> tags, Predicate<Person> predicate,
 
     private List<Boolean> getValidParameters(Person person) {
         List<Boolean> validParameters = new ArrayList<>();
-        addIfPresent(() -> toFind.getName().toString(), p -> {
-            String[] nameKeywords = toFind.getName().toString().split("\\s+");
+        addIfPresent("name", p -> {
+            String[] nameKeywords = inputParameters.get("name").toString().split("\\s+");
             NameContainsKeywordsPredicate namePredicate =
                     new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords));
             return namePredicate.test(p);
         }, person, validParameters);
 
-        addIfPresent(() -> toFind.getPhone().toString(), p -> toFind.getPhone().equals(p.getPhone()),
+        addIfPresent("phone", p -> inputParameters.get("phone").equals(p.getPhone()),
                 person, validParameters);
 
-        addIfPresent(() -> toFind.getEmail().toString(), p -> toFind.getEmail().equals(p.getEmail()),
+        addIfPresent("email", p -> inputParameters.get("email").equals(p.getEmail()),
                 person, validParameters);
 
-        addIfPresent(() -> toFind.getAddress().toString(), p -> toFind.getAddress().equals(p.getAddress()),
+        addIfPresent("address", p -> inputParameters.get("address").equals(p.getAddress()),
                 person, validParameters);
 
-        addIfPresent(() -> toFind.getProjectStatus().toString(), p -> toFind.getProjectStatus()
+        addIfPresent("project status", p -> inputParameters.get("project status")
                 .equals(p.getProjectStatus()), person, validParameters);
 
-        addIfPresent(() -> toFind.getPaymentStatus().toString(), p -> toFind.getPaymentStatus()
+        addIfPresent("payment status", p -> inputParameters.get("payment status")
                     .equals(p.getPaymentStatus()), person, validParameters);
 
-        addIfPresent(() -> toFind.getClientStatus().toString(), p -> toFind.getClientStatus()
+        addIfPresent("client status", p -> inputParameters.get("client status")
                     .equals(p.getClientStatus()), person, validParameters);
 
-        addIfDeadlinePresent(() -> toFind.getDeadline().toString(), p -> toFind.getDeadline()
+        addIfPresent("deadline", p -> inputParameters.get("deadline")
                 .equals(p.getDeadline()), person, validParameters);
 
-        addIfTagPresent(toFind.getTags(), p -> toFind.getTags().stream()
+        @SuppressWarnings("unchecked")
+        Set<Tag> tags = (Set<Tag>) inputParameters.get("tags");
```

Comment:
```
@SuppressWarnings("unchecked") 
Set<Tag> tags = (Set<Tag>) inputParameters.get("tags");
addIfTagPresent(tags, p -> tags.stream()
    .anyMatch(searchTag -> p.getTags().stream()
        .anyMatch(personTag -> personTag.tagName.equalsIgnoreCase(searchTag.tagName))), 
    person, validParameters);
return validParameters;
```
To make it case-insensitive we can try equalsIgnoreCase Instead of using direct contains (p.getTags()::contains)
what do you think?

Reply:
you are right about tag not being case insensitive, I will change the equality comparison of tags in the tag class instead